shell/platform/android/io/flutter/plugin/editing/ImeSyncDeferringInsetsCallback.java

```diff
@@ -131,24 +126,24 @@ public WindowInsets onProgress(
       if (!matching) {
         return insets;
       }
+
+      // The IME insets include the height of the navigation bar. If the app isn't laid out behind
+      // the navigation bar, this causes the IME insets to be too large during the animation.
+      // To fix this, we subtract the navigationBars bottom inset if the system UI flags for laying
+      // out behind the navigation bar aren't present.
+      int excludedInsets = 0;
+      int systemUiFlags = view.getWindowSystemUiVisibility();
+      if ((systemUiFlags & View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION) == 0
```

Comment:
Without committing you to building it do you think we need to do a similar evaluation for status bar and the top insets?

Reply:
I don't think that'll be necessary as the status bar visibility and top insets didn't affect the keyboard animation in my testing.

Reply:
But it would impact any animation coming from the top like keyboards impact animations coming from the bottom. Right? If for example the status bar was present our code would behave differently than if status bar was overlayed. I am thinking of https://github.com/flutter/flutter/issues/118761

Reply:
@gmackall this is something to keep in mind.